client/my-sites/customer-home/cards/tasks/index.jsx

```diff
@@ -0,0 +1,121 @@
+import Gridicon from 'components/gridicon';
+import QueryPublicizeConnections from 'components/data/query-publicize-connections';
+import { savePreference } from 'state/preferences/actions';
+import { getPreference } from 'state/preferences/selectors';
+import { getSelectedSiteId, getSelectedSiteSlug } from 'state/ui/selectors';
+import { getConnectionsBySiteId } from 'state/sharing/publicize/selectors';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+class RecommendedTask extends Component {
+	render() {
+		const { hideSocialTask, siteId, siteSlug, translate } = this.props;
+
+		if ( hideSocialTask ) {
+			return null;
+		}
+
+		let taskName;
+		let taskId;
+		let taskImage;
+		let taskDescription;
+		let taskMinutes;
+		let actionText;
+		let actionLink;
+		let shouldQueryPublicizeConnections;
+
+		if ( ! hideSocialTask ) {
+			taskName = translate( 'Connect social media accounts' );
+			taskId = 'SocialMedia';
+			taskImage = '/calypso/images/stats/tasks/social-links.svg';
+			taskDescription = translate(
+				'Integrate your site with social media to automatically post your content and drive traffic to your site.'
+			);
+			taskMinutes = '3';
+			actionText = translate( 'Connect accounts' );
+			actionLink = isMobile()
+				? `/marketing/connections/${ siteSlug }`
+				: `/marketing/connections/${ siteSlug }?tour=marketingConnectionsTour`;
+			shouldQueryPublicizeConnections = true;
+		}
+
+		const minuteOrMinutes =
+			taskMinutes > 1
+				? translate( 'minutes', {
+						comment: 'eg. 5 minutes to complete a task',
+				  } )
+				: translate( 'minute', {
+						comment: 'eg. 1 minute to complete a task',
+						context: 'noun',
+				  } );
+		return (
+			<ActionPanel className="tasks">
+				{ shouldQueryPublicizeConnections && <QueryPublicizeConnections selectedSite /> }
+				<ActionPanelBody>
+					{ ! isMobile() && (
+						<ActionPanelFigure align="right">
+							<img src={ taskImage } alt="" />
+						</ActionPanelFigure>
+					) }
+					<div className="tasks__timing">
+						<Gridicon icon="time" size={ 18 } />
+						<p>{ taskMinutes + ' ' + minuteOrMinutes }</p>
+					</div>
+					<ActionPanelTitle>{ taskName }</ActionPanelTitle>
+					<p className="tasks__description">{ taskDescription }</p>
+					<ActionPanelCta>
+						<Button className="tasks__action" primary href={ actionLink }>
+							{ actionText }
+						</Button>
+						<Button
+							className="tasks__action-skip is-link"
+							onClick={ () => {
+								this.props.skipTask( 'homeTask' + taskId + siteId, true );
+							} }
+						>
+							{ translate( 'Skip for now' ) }
+						</Button>
+					</ActionPanelCta>
+				</ActionPanelBody>
+			</ActionPanel>
+		);
+	}
+}
+
+const mapStateToProps = state => {
+	const siteId = getSelectedSiteId( state );
+
+	return {
+		siteId,
+		siteSlug: getSelectedSiteSlug( state ),
+		hideSocialTask:
+			getPreference( state, 'homeTaskSocialMedia' + siteId ) ||
+			getConnectionsBySiteId( state, siteId ).length > 0,
```

Comment:
Feel free to drop this, the client cards in My Home can be super straightforward ✅Eg if the server tells us to render it, it's fine.

We can help with marking this as done on the server ✨ 

Speaking of which @mmtr @kwight we should think though if we like the server checklist pattern or if should come up with something better for marking tasks as done.

Reply:
Should I drop the `getPreference  as well or just the `getConnectionsBySiteId`? :)

Reply:
We can drop all visibility checks for My Home client cards. The card can of course still fetch the data it needs to render though 👍 

The idea is that the client should be able to trust everything that the layout API says to render (we do a bunch of visibility checks on the server side for each card to avoid a number of API requests.) So in the future the response might have an extra "tasks" property that has a list of task cards to render.
<img width="1390" alt="Screen Shot 2020-04-10 at 2 35 57 PM" src="https://user-images.githubusercontent.com/1270189/79024713-a4b60a00-7b38-11ea-93e6-4db83f496ae2.png">

Reply:
I removed the logic to hide the card, but I kept the `savePreference` since I'm assuming that'll still be necessary. :)